lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -41,6 +41,10 @@ def query_api(
         return query_api
 
     def mock_find_dataset(self, dataset):
+        ds = Dataset.query(resource_id=dataset)
+        if ds is None:
+            raise DatasetNotFound(resource_id=ds)
```

Comment:
`Dataset.query` already does this; if `Dataset.query` returns, `ds` isn't `None`.

In any case, your exception is misleading as you're using the value `None` for `resource_id` rather than the `dataset` value you tried to find. The real `raise` inside `Dataset.query` uses `**kwargs`.

Try:

```
    def mock_find_dataset(self, dataset):
        class Tarball(object):
            unpacked = Path("/dataset1/")

        # Validate the resource_id
        Dataset.query(resource_id=dataset)
        return Tarball
```

You don't even need the return value since you're only interested in whether it raises `DatasetNotFound`.

Reply:
Yeah sure @dbutenhof.This looks pretty optimized, will make these minor updates and push the changes :+1: